libs/coroutines/src/detail/coroutine_self.cpp

```diff
@@ -14,7 +14,7 @@
 namespace hpx { namespace threads { namespace coroutines { namespace detail {
     coroutine_self*& coroutine_self::local_self()
     {
-        HPX_NATIVE_TLS coroutine_self* local_self_ = nullptr;
```

Comment:
Hah, this was a bug to begin with! I wonder how it ever worked?

Reply:
According to the standard it is:
>When thread_local is applied to a variable of block scope the storage-class-specifier static is implied if it does not appear explicitly.

See, and now I have learned some thing new. I think we should have the `static` explicitly, however.